### PR TITLE
YPE-1141: Add loading spinner to BibleVersionPicker

### PR DESCRIPTION
**Pull request by Kyleasmth**
## Summary
- Show a loading spinner in the version list and language badge while versions are fetching, instead of flashing "No versions found"
- Add unit tests covering loading, empty, and loaded states
- Add a Storybook Loading story for visual QA
## Test plan
- Verify spinner displays in version list area while loading
- Verify spinner displays in the language button badge while loading
- Verify "No versions found" only appears after loading completes with no results
 Run pnpm --filter @youversion/platform-react-ui test
 
 
Screenshot 2026-02-19 at 8 16 21 AM

### CODE REVIEW

**Review by cameronapak — CHANGES_REQUESTED** [review]

File: packages/ui/src/components/bible-version-picker.tsx [hunk 1]
```diff
@@ -524,7 +532,11 @@ function Content() {
...
             <div className="yv:w-full yv:flex yv:items-center yv:justify-center yv:py-8 yv:text-center yv:text-muted-foreground yv:text-sm">
-              No versions found
+              {versionsLoading ? (
+                <LoaderIcon className="yv:size-6 yv:animate-spin yv:text-muted-foreground" />
+              ) : (
+                'No versions found'
+              )}
             </div>
           ) : null}
         </div>
```
[the remark this answers is not in the document]
Comment (cameronapak) [reply packages/ui/src/components/bible-version-picker.tsx]: PErfect!

**Review by cameronapak — APPROVED** [review]
Works like a charm. Approved. Please see if you need to make any changes based on Jared's PR comment

**Review by jaredhightower-youversion — APPROVED** [review]
